<br/>
Since there are many robust and capable servers for functional enrichment
analysis, we feel it is better to use a previously tool/web server than to 
try to reproduce what is already avaiable.

Examples of these services include the following:
  * [DAVID](https://david.ncifcrf.gov/)
  * [UCSC Genome Browser](https://genome.ucsc.edu/)
  * [UniProt](https://www.uniprot.org/)

For this example, we will look at how you can extract differentially 
expressed gene data for IRIS-EDA and how you can use DAVID for functional
enrichment.

1. Once you have run **DEG Analysis** in the `DEG Analysis` tab, simply
   download the differentially expressed genes using the 
   `Download Filtered Data` button on the prior tab (`Plots` tab).
   
2. This will download a set of *significantly* differentially expressed genes 
   or IDs that can be opened up in R, Excel, LibreOffice Calc, or any other
   spreadsheet viewing software.
   
3. Once the spreadsheet is open, simply copy the IDs from the first column and
   open up DAVID Bioinformatics Resources (https://david.ncifcrf.gov/).
   
4. 
